libmachine/provision/redhat.go

```diff
@@ -0,0 +1,235 @@
+package provision
+
+import (
+	"bytes"
+	"fmt"
+	"text/template"
+
+	"github.com/docker/machine/drivers"
+	"github.com/docker/machine/libmachine/auth"
+	"github.com/docker/machine/libmachine/engine"
+	"github.com/docker/machine/libmachine/provision/pkgaction"
+	"github.com/docker/machine/libmachine/swarm"
+	"github.com/docker/machine/log"
+	"github.com/docker/machine/utils"
+)
+
+const (
+	// TODO: eventually the RPM install process will be integrated
+	// into the get.docker.com install script; for now
+	// we install via vendored RPMs
+	dockerRPMPath = "https://docker-mcn.s3.amazonaws.com/public/redhat/rpms/docker-engine-1.6.1-0.0.20150511.171646.git1b47f9f.el7.centos.x86_64.rpm"
+)
+
+func init() {
+	Register("RedHat", &RegisteredProvisioner{
+		New: NewRedHatProvisioner,
+	})
+}
+
+func NewRedHatProvisioner(d drivers.Driver) Provisioner {
+	return &RedHatProvisioner{
+		GenericProvisioner{
+			DockerOptionsDir:  "/etc/docker",
+			DaemonOptionsFile: "/lib/systemd/system/docker.service",
```

Comment:
Yeah, this is definitely not a place we should be writing to; this should definitely go in `/etc/systemd/system/docker.service` instead (as a stub, not the full file, as I described in https://github.com/docker/machine/pull/1090/files#r29536500).